--- a/mksquashfs.c	2025-11-03 12:47:59.847470992 +0100
+++ b/mksquashfs.c.patches	2025-11-03 12:51:59.455832390 +0100
@@ -47,6 +47,7 @@
 #include <regex.h>
 #include <fnmatch.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 
 #ifndef linux
 #define __BYTE_ORDER BYTE_ORDER
@@ -1285,25 +1286,25 @@
 	}
 	else if(type == SQUASHFS_CHRDEV_TYPE || type == SQUASHFS_BLKDEV_TYPE) {
 		squashfs_dev_inode_header *dev = &inode_header.dev;
-		unsigned int major = major(buf->st_rdev);
-		unsigned int minor = minor(buf->st_rdev);
+		unsigned int _major = major(buf->st_rdev);
+		unsigned int _minor = minor(buf->st_rdev);
 
-		if(major > 0xfff) {
+		if(_major > 0xfff) {
 			ERROR("Major %d out of range in device node %s, "
-				"truncating to %d\n", major, filename,
-				major & 0xfff);
-			major &= 0xfff;
+				"truncating to %d\n", _major, filename,
+				_major & 0xfff);
+			_major &= 0xfff;
 		}
-		if(minor > 0xfffff) {
+		if(_minor > 0xfffff) {
 			ERROR("Minor %d out of range in device node %s, "
-				"truncating to %d\n", minor, filename,
-				minor & 0xfffff);
-			minor &= 0xfffff;
+				"truncating to %d\n", _minor, filename,
+				_minor & 0xfffff);
+			_minor &= 0xfffff;
 		}
 		inode = get_inode(sizeof(*dev));
 		dev->nlink = nlink;
-		dev->rdev = (major << 8) | (minor & 0xff) |
-				((minor & ~0xff) << 12);
+		dev->rdev = (_major << 8) | (_minor & 0xff) |
+				((_minor & ~0xff) << 12);
 		SQUASHFS_SWAP_DEV_INODE_HEADER(dev,
 			(squashfs_dev_inode_header *) inode);
 		TRACE("Device inode, rdev 0x%x, nlink %d\n", dev->rdev, nlink);
@@ -3177,7 +3178,7 @@
 }
 
 
-static inline void add_dir_entry(char *name, char *pathname, struct dir_info *sub_dir,
+inline void add_dir_entry(char *name, char *pathname, struct dir_info *sub_dir,
 	struct inode_info *inode_info, struct dir_info *dir)
 {
 	if((dir->count % DIR_ENTRIES) == 0) {
